astropy/coordinates/representation.py

```diff
@@ -1909,6 +1988,32 @@ def from_cartesian(cls, cart):
         # erfa p2s: P-vector to spherical polar coordinates.
         return cls(*erfa_ufunc.p2s(p), copy=False)
 
+    def transform(self, matrix):
+        """Transform the spherical coordinates using a 3x3 matrix.
+
+        This returns a new representation and does not modify the original one.
+        Any differentials attached to this representation will also be
+        transformed.
+
+        Parameters
+        ----------
+        matrix : (3,3) array-like
+            A 3x3 (or stack thereof) matrix, such as a rotation matrix.
+
+        """
+        if self.differentials:
+            # TODO! shortcut if there are differentials.
+            # Currently just super, which uses Cartesian backend.
+            rep = super().transform(matrix)
+
+        else:
+            xyz = erfa_ufunc.s2c(self.lon, self.lat)
+            p = erfa_ufunc.rxp(matrix, xyz)
+            lon, lat, ur = erfa_ufunc.p2s(p)
+            rep = self.__class__(lon=lon, lat=lat, distance=self.distance * ur)
```

Comment:
Duplicated code and docstring is making me wonder if this can be abstracted anywhere... Maybe not for this PR, but worth a thought.